src/shared/makeImageSizes.ts

```diff
@@ -0,0 +1,14 @@
+export type ImageWidthsWithBreakpointsProps = {
+  small: string;
+  medium: string;
+  large: string;
+};
+
+export default function makeImageSizes({
+  small,
```

Comment:
I thnk we can remove this file by using some tailwind methods for media query
- https://tailwindcss.com/docs/responsive-design

Reply:
I'll answer all the comments related to this issue here.

When we use the next `<Image>` we either use props `height` and `width` with some CSS on the `Image` itself or parent element, or we use pros `fill` and `sizes` and some CSS. This is in the docs and if you don't do that you get a console log dev tool warning about it.

That being said. I tried  both ways in this project and in some other personal projects and realized the method where we use props `fill` and `sizes` is the best one when it comes to styling the `Image` with no weird side effects.

It's weird but after you set it to `fill` you need to use the `sizes` and object fit contain - see docs. `sizes` takes in an array with media queries like the one returned by the `makeImageSizes` function, I figured out we might need this later and didn't want to have type such long strings everytime we need to use the `Image` with the `fill` + `sizes` method.

I'll change the sizes. I forgot to check that.

Reply:
Usually with next images i use them like this

```js
<Wrapper style={{maxWidth:500px,maxHeight:400px,display:flex,flex:1}}>
<Image src={src} alt={alt} width={MaxWidthIMightNeed} height={MaxHeightIMightNeed}
styles={{width:100%,height:100%,objectFitCover}}
/>
</Wrapper>

```

Reason: when it comes to width, height you pass to the Image component itself, it actually tells next server side, to preprocess  the image to fit that spec, so that you don't send images that are too large to the page, so it basically controls the resolution not its behaviour on the layout.

 To control the behaviour on the layout i usually wrap it in a div that fits my spec and set width,height inside the style tag on the img component to 100%.
 
 The advantage here is that you don't send images that are too big (in mb) to the page
 
 > it's also usually good to set different width,height for mobile so that the resolution is smaller thus the image can actually load on a 3g connection

Reply:
Hmm, let's do a pair programming o this. It's usually a bad idea to calculate CSS stuff inside JS. Let's see if we can find a workaround 💪🏼

Reply:
@vitorschelb